addons/mail/static/src/core/message_service.js

```diff
@@ -249,7 +249,7 @@ export class MessageService {
                 ? message.starred_partner_ids.includes(this.store.user.id)
                 : false,
             isTransient,
-            parentMessage: message.parentMessage ? this.insert(message.parentMessage) : undefined,
+            parentMessage: data.parentMessage ? this.insert(data.parentMessage) : undefined,
```

Comment:
@seb-odoo this solves an infinite loop when showing the pinned panel when it contains a reply-to message.

The infinite loop is specific to master, from moving pin code to its dedicated folder. `getPinnedMessages` does `messageService.insert()`. which triggers this code, that re-renders template, etc.

I'll make the same fix in 16.3

Reply:
Same issue I had in other cases,

Reply:
But shouldn't `parentMessage` be extracted to avoid having it in `remainingData`?
Otherwise it will be overwritten once by data and then again by record, even if it was not supposed to change.